client/Client.cpp

```diff
@@ -209,41 +209,45 @@ void CClient::loadGame(CGameState * initializedGameState)
 		logNetwork->info("Game state was transferred over network, loading.");
 		gs = initializedGameState;
 	}
-	else
+
+	// assignment of the loader, which is needed to deserialze data such as sleepingHeroes
+	try
 	{
-		try
-		{
-			boost::filesystem::path clientSaveName = *CResourceHandler::get("local")->getResourceName(ResourceID(CSH->si->mapname, EResType::CLIENT_SAVEGAME));
-			boost::filesystem::path controlServerSaveName;
+		boost::filesystem::path clientSaveName = *CResourceHandler::get("local")->getResourceName(ResourceID(CSH->si->mapname, EResType::CLIENT_SAVEGAME));

```

Comment:
Feels like you want to merge gamestate passed over network (gs) and thing loaded from file. You removed else and now initalizedGamestate is not used. As far as I understand things. So you can try to load from file first and put needed data into initializedGamestate and gs = initializedGamestate at the end.

Anyway it is more theoretical, lets wait somebody more acquainted in this area.

Reply:
I have also tried to reorder the data loading but it still crashes with similar complain about `primaryID`.

Currently I am trying to workaround this by reading the data from file to another object and add sleepHeroes back to the game based on some identifiers (`heroId`?), but I still have to understand how the serialization actually works to implement it.